bigquery_etl/shredder/config.py

```diff
@@ -544,18 +549,15 @@ def __get_tables_with_pioneer_id__(dataset_list):
             pioneer_target(
                 table=qualified_table_id(table), project=PIONEER_PROD
             ): sources[table.dataset_id]
-            for table in __get_tables_with_pioneer_id__(derived_datasets)
+            for dataset in derived_datasets
+            for table in __get_tables_with_pioneer_id__(dataset)
         },
         **{
             # tables with pioneer_id located in study analysis projects
             pioneer_target(
                 table=qualified_table_id(table), project=table.project
-            ):  # The following finds corresponding delete request table.
-            # By convention, it is located under stable study dataset in
-            # PIONEER_PROD project
-            sources[
-                re.sub("^moz-fx-data-", "", table.project).replace("-", "_") + "_stable"
-            ]
-            for table in __get_tables_with_pioneer_id__(analysis_datasets)
+            ): sources[study + "_stable"]
```

Comment:
Study names aren't normalized so this might need to include the `.replace("-", "_")` that was present in the previous logic.

Reply:
Thanks, included this.